File: problems/4558bf52-fdb6-41f8-b3c0-e7607c170642/go/driver.go
package main

import (
	"encoding/json"
	"log"
	"os"
)

type Testcase struct {
	Id       int  `json:"id"`
	Input    CustomInput  `json:"input"`
	Expected [][]int  `json:"expected"`
	IsPublic bool `json:"isPublic"`
}

type CustomInput struct {
	List []int `json:"list"`
	K    int   `json:"k"`
}

func loadTestcases(filename string) []*Testcase {
	file, err := os.Open(filename)
	if err != nil {
		log.Fatal("failed to open testcases file.")
	}

	var testcases []*Testcase

	decoder := json.NewDecoder(file)
	err = decoder.Decode(&testcases)
	if err != nil {
		log.Fatal("failed to decode testcases file.")
	}

	return testcases
}

File: problems/461fe85c-4e7d-41a5-97ba-cd8d526aa632/typescript/driver.ts
import * as fs from 'fs';

interface Testcase {
    id: number;
    input: string;
    expected: boolean;
    isPublic: boolean;
}

export function loadTestcases(filename: string): Testcase[] {
    try {
        const data = fs.readFileSync(filename, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        console.error('Failed to open testcases file.');
        process.exit(1);
    }
}

export type { Testcase };
